task/upload-sbom-to-trustification/0.1/upload-sbom-to-trustification.yaml

```diff
@@ -0,0 +1,231 @@
+    image: quay.io/redhat-appstudio/appstudio-utils:5bd7d6cb0b17f9f2eab043a8ad16ba3d90551bc2@sha256:8c7fcf86af40c71aeb58e4279625c8308af5144e2f6b8e28b0ec7e795260e5f7
+    script: |
+      #!/usr/bin/env bash
+      set -o errexit -o nounset -o pipefail
+
+      shopt -s nullglob
+      sboms_to_upload=("$WORKDIR"/*.json)
+
+      if [[ "${#sboms_to_upload[@]}" -eq 0 ]]; then
+        echo "No SBOMs to upload"
+        exit 0
+      fi
+
+      read_required_secret_key() {
+        local key="$1"
+        if [[ -f "$TRUSTIFICATION_SECRET_PATH/$key" ]]; then
+          cat "$TRUSTIFICATION_SECRET_PATH/$key"
+        else
+          echo "Missing configuration: $key" >&2
+          echo "Does the '$TRUSTIFICATION_SECRET_NAME' secret exist in your namespace and contain the required keys?" >&2
+          echo "Refer to the description of this Task for details." >&2
+          exit 1
+        fi
+      }
+
+      bombastic_api_url="$(read_required_secret_key bombastic_api_url)"
+      oidc_issuer_url="$(read_required_secret_key oidc_issuer_url)"
+      oidc_client_id="$(read_required_secret_key oidc_client_id)"
+      oidc_client_secret="$(read_required_secret_key oidc_client_secret)"
+
+      curl_opts=(--silent --show-error --fail-with-body --retry "$HTTP_RETRIES")
+
+      # https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfig
+      openid_configuration_url="${oidc_issuer_url%/}/.well-known/openid-configuration"
+      echo "Getting OIDC issuer configuration from $openid_configuration_url"
+      # https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata
+      token_endpoint="$(curl "${curl_opts[@]}" "$openid_configuration_url" | jq -r .token_endpoint)"
+
+      # https://www.rfc-editor.org/rfc/rfc6749.html#section-4.4.2
+      # https://www.rfc-editor.org/rfc/rfc6749.html#section-2.3.1
+      basic_auth="$(
+        jq -r -n --arg username "$oidc_client_id" --arg password "$oidc_client_secret" \
+          '($username | @uri) + ":" + ($password | @uri) | @base64'
+      )"
+
+      for sbom_path in "${sboms_to_upload[@]}"; do
+        original_sbom_relpath="$(realpath "$sbom_path" --relative-base="$SBOMS_DIR")"
+        echo
+        echo "--- Processing $original_sbom_relpath ---"
+
+        echo "Getting OIDC token from $token_endpoint"
+        token_response="$(
+          curl "${curl_opts[@]}" \
+            -H "authorization: Basic $basic_auth" \
+            -d "grant_type=client_credentials" \
+            "$token_endpoint"
+        )"
+        # https://www.rfc-editor.org/rfc/rfc6749.html#section-5.1
+        access_token="$(jq -r .access_token <<< "$token_response")"
+        token_type="$(jq -r .token_type <<< "$token_response")"
+        expires_in="$(jq -r ".expires_in // empty" <<< "$token_response")"
+
+        retry_max_time=0  # no limit
+        if [[ -n "$expires_in" ]]; then
+          retry_max_time="$expires_in"
+        fi
+
+        sbom_id="$(basename -s .json "$sbom_path")"
```

Comment:
I think you want something more unique here. Maybe `uuidgen`?

Reply:
The id is the sha256 checksum of the SBOM content (before downgrading its CycloneDX version). The idea is that this avoids duplication when the same SBOM is uploaded multiple times.

(It's not very clear from the code though, so I'll add a comment to clarify).